internal/rbd/controllerserver.go

```diff
@@ -158,6 +158,44 @@ func (cs *ControllerServer) parseVolCreateRequest(ctx context.Context, req *csi.
 	return rbdVol, nil
 }
 
+// getVolumeFromID gets the rbd image details from the volumeID and also takes
+// volume lock and creates the new user credential.
+// we need to call cs.cleanup() in defer to remove the lock and credentials.
+func (cs *ControllerServer) getVolumeFromID(ctx context.Context, volumeID string, secrets map[string]string) (*rbdVolume, *util.Credentials, error) {
+	// validate the volume ID
+	cr, err := util.NewUserCredentials(secrets)
+	if err != nil {
+		return nil, nil, status.Error(codes.Internal, err.Error())
```

Comment:
I wonder if you need to return a gRPC error here. The callers of `cs.getVolumeID()` seem to take the error, inspect it and return their own gRPC version.

Reply:
Yes removed GRPC errors from this function

Reply:
uh, line 173: https://github.com/ceph/ceph-csi/pull/1924/commits/f9fb9b6cd260e40cb54d0d6386bfb6a4b92c2604#diff-3792fcc7313c4d1e0b81c4d3ff7ac714b9d299130641673729196b2a4a1d6f2aR173